File: java/Medium/_456_132Pattern.java
package Medium;

import java.util.Stack;

public class _456_132Pattern {
    public boolean find132pattern(int[] nums) {
        Stack<Integer> stk = new Stack<>();
        int third = Integer.MIN_VALUE;
        for (int i = nums.length-1; i >=0; i--){
            if(nums[i] < third) return true;
            while(!stk.isEmpty() && nums[i] > stk.peek()) third = stk.pop();
            stk.push(nums[i]);
        }
        return false;
    }
}

File: Python3/_456_132Pattern.py
from typing import List


class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        third = -1000000000
        stack = []
        for i in range(len(nums)-1,-1,-1):
            if nums[i] < third:
                return True

            while len(stack) > 0 and nums[i] >stack[-1]:
                third = stack.pop()

            stack.append(nums[i])

        return False